Stepper accessibility (#4682)
kiwicom/orbit: docs/src/documentation/03-components/07-interaction/stepper/03-accessibility.mdx

- The `titleIncrement` prop allows you to specify an `aria-label` attribute for the increment icon button in the Stepper (StepperStateless) component.

- The `descriptionDecrement` prop allows you to specify an `aria-describedby` attribute for the decrement icon button in the Stepper (StepperStateless) component.

- The `descriptionIncrement` prop allows you to specify an `aria-describedby` attribute for the increment icon button in the Stepper (StepperStateless) component.

- The `ariaLabelledBy` prop allows you to specify an `aria-labelledby` attribute for the input field in the Stepper component. This attribute references the ID of the element that labels the Stepper (StepperStateless), ensuring that screen readers announce the label correctly.

> - The `ariaDescribedby` prop allows you to specify an `aria-describedby` attribute for the input field in the Stepper component. This attribute references the ID of the element that describes the Stepper (StepperStateless), providing additional context or instructions.

Reviewer: I would group here props for input vs prop for buttons so it's easier for the reader to make sense of it.

And I would unify the information, e.g. for `ariaDescribedby` you mention what the attribute references but you don't do it for `descriptionDecrement`. Sometimes input element is mentioned, sometimes input field.

Would it make sense to maybe put these props in the table and write some common text under the group of elements?

- The `titleIncrement` prop allows you to specify an `aria-label` attribute for the increment icon button in the Stepper (StepperStateless) component.

> - The `ariaLabelledBy` prop allows you to specify an `aria-labelledby` attribute for the Stepper component. This attribute references the ID of the element that labels the Stepper (StepperStateless), ensuring that screen readers announce the label correctly.

Reviewer: If you will keep the `ariaLabelledBy`, I would add here that it's labelling just the value, not buttons or the whole component, just to be more clear.

```jsx
<Stack>
  <Stack>
    <Text id="adults">Adults</Text>
  </Stack>
  <Stepper
    step={1}
    minValue={0}
    minValue={10}
    ariaLabelValue="Number of passengers"
    ariaDescribedby="adults"
    titleDecrement="Remove a passenger"
    titleIncrement="Add a passenger"
  />
</Stack>
```

> This example includes `ariaDescribedby` prop. Value of `ariaDescribedby` prop is used to announce the complementary information about the Stepper component title (`Adults`) once the Stepper button is focused by the screen reader.

Reviewer: Here, as you added `aria-describedby` for both buttons and input, I guess it makes sense to mention it - because it's not announcing complementary information only when Stepper button is focused, but also other two elements.

```jsx
<Stack>
  <Stack>
    <Text id="adults-title">Adults</Text>
    <Text id="adults-description">Number of adult passengers</Text>
  </Stack>
  <Stepper
    step={1}
    minValue={0}
    minValue={10}
    ariaLabelValue="Number of passengers"
    ariaLabelledBy="adults-title"
    ariaDescribedby="adults-description"
    titleDecrement="Remove a passenger"
    titleDecrement="Add a passenger"
    descriptionIncrement="adults-description"
    descriptionDecrement="adults-description"
  />
</Stack>
```

This example includes `ariaLabelledby` prop. In this case, `ariaLabelledBy` prop is prioritized over `ariaLabelValue`, so the screen reader will announce the value title (`Adults`) and buttons title (`Add a passenger`, `Remove a passenger`) once they are focused by the screen reader.

> Value of `ariaDescribedby`, `descriptionIncrement` and `descriptionDecrement` prop is used to announce the complementary information about the Stepper component description (`Number of adults in your group`) once the Stepper input or buttons are focused by the screen reader.

Reviewer: I think this example is a bit of a mess.

E.g. in the last sentence you mention "Number of adults in your group" but that's not in the example.
Maybe for `descriptionIncrement` and `descriptionDecrement` you wanted to add "adults-title"? Because I don't think that "Add a passenger. Number of adult passengers." is correct 😅 
You also sometimes have `ariaLabelledby` and sometimes `ariaLabelledBy`.
Also I would either select different text because I don't think it makes sense to say for the value "Aduls. Number of adult passengers."

And overall I think it would make sense to write it in order how it's read on focus. So first mention everything that is read for decrement button, then value, then increment button.

</Stack>
```

This example includes `ariaLabelledby` prop. In this case, `ariaLabelledby` prop is prioritized over `ariaLabelValue`, so the screen reader will announce the value title (`Passengers`) and buttons title (`Add a passenger`, `Remove a passenger`) once they are focused by the screen reader.

```jsx
<Stack>
  <Stack>
    <Text id="adults-title">Adults</Text>

>     <Text id="adults-description">Number of passengers in your group</Text>

Reviewer: I think you forgot to rewrite this to "Number of adults in your group"

  </Stack>
  <Stepper
    step={1}
    minValue={0}

>     minValue={10}

Reviewer: `maxValue`, applies to all examples.

Author: yes, leftover, thx.

    ariaLabelValue="Number of passengers"
    ariaLabelledby="adults-title"
    ariaDescribedby="adults-description"
    titleDecrement="Remove a passenger"
    titleIncrement="Add a passenger"
    descriptionIncrement="adults-title"
    descriptionDecrement="adults-title"
  />
</Stack>
```

This example includes `ariaLabelledby`, `descriptionIncrement` and `descriptionDecrement` props.

When decrement button is focused by screen reader, the screen reader will announce the button title (`Remove a passenger`) and complementary information (`Adults`).

> For input field, the value of `ariaLabelValue` (`Number of passengers`) is read first, then the complementary information (value of `ariaDescribedby` prop) is announced - `Number of adults in your group`. Value of `ariaLabelledby` is ignored in this case.

Reviewer: Why is value of `ariaLabelledby` ignored if it's the same case as in the previous example?

> The same logic as for decrement button is applicable for the increment button. When increment button is focused by screen reader, the screen reader will announce the button title (`Add a passenger`) and complementary information (`Adults`).

Reviewer: ```suggestion
The same logic applied to the decrement button applies to the increment button. When increment button is focused by screen reader, the screen reader will announce the button title (`Add a passenger`) and complementary information (`Adults`).
```
